frontend/src/contexts/modalContext.tsx

```diff
@@ -0,0 +1,53 @@
+import { ModalProps } from 'components/common/GenericModal';
+import noop from 'lodash/noop';
+import * as React from 'react';
+import { useState } from 'react';
+
+export interface IModalContext {
+  modalProps?: ModalProps;
+  setModalProps: (modalProps?: ModalProps) => void;
+  setDisplayModal: (display: boolean) => void;
+}
+
+export const ModalContext = React.createContext<IModalContext>({
+  modalProps: undefined,
+  setModalProps: noop,
+  setDisplayModal: noop,
+});
+
+export const ModalContextProvider = (props: IModalContext & { children: any }) => {
```

Comment:
Thoughts on limiting the children to be a JSX or string instead of any?

Reply:
Yes, there is actually a reactChild type that I can use here, more copy pasta